### PR TITLE
User can see a list of all their favorites

### LINKED ISSUE #8: User Story - User Can View All Their Favorites
As a user,
When I GET to `/api/v1/favorites`,
I should get a JSON response with the following format

**Successful response**
Status code 200
```json
[
  {
    "id": 1,
    "title": "We Will Rock You",
    "artistName": "Queen",
    "genre": "Rock",
    "rating": 88
  },
  {
    "id": 2,
    "title": "Careless Whisper",
    "artistName": "George Michael",
    "genre": "Pop",
    "rating": 93
  },
]
```

### PR DESCRIPTION
**Pull request by CoopTang**
* Refactor project tree to resemble MVC paradigm
* Add FavoritesController to route to the index for all Favorites
* Rename Favorite column name from snake_case to camelCase

closes #8

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -17,5 +18,6 @@ app.use(cookieParser());
 app.use(express.static(path.join(__dirname, 'public')));
 
 app.use('/', indexRouter);
+app.use('/api/v1/favorites', favoritesRouter);
```
Comment (Jonpatt92) [inline app.js]: Good router usage

File: lib/controllers/favorites_controller.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+const index = (request, response) => {
+  Favorite.all()
+    .then((favorites) => {
+      response.status(200).json(favorites)
+    })
+    .catch((error) => {
+      response.status(500).json({ error });
+    })
+}
```
Comment (Jonpatt92) [inline lib/controllers/favorites_controller.js]: Good encapsulation of index method

File: lib/models/favorite.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+const environment = process.env.NODE_ENV || 'development';
+const configuration = require('../../knexfile')[environment];
+const database = require('knex')(configuration);
+
+const all = () => database('favorites')
+  .select()
```
Comment (Jonpatt92) [inline lib/models/favorite.js]: Great model setup

File: tests/requests/api/v1/favorites/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+  beforeEach(async () => {
+    await database.raw('truncate table favorites cascade');
+
+    let favoriteData = {
+      "title": "We Will Rock You",
+      "artistName": "Queen",
+      "genre": "Rock",
+      "rating": 88
+    }
+    await database('favorites').insert(favoriteData, 'id');
```
Comment (Jonpatt92) [inline tests/requests/api/v1/favorites/index.spec.js]: Great creation of records for testing. We will probably need one more Favorite record to truly verify we're being returned all records.

**Comment by CoopTang** [comment]
> Add another favorite record to the 'favorites index' endpoint test

Added an additional item in the Favorite index test.

File: tests/requests/api/v1/favorites/index.spec.js [hunk 5]
```diff
@@ -40,6 +48,16 @@ describe('Test the favorites path', () => {
...
+
+      expect(res.body[1]).toHaveProperty('id');
+      expect(res.body[1]).toHaveProperty('title');
+      expect(res.body[1].title).toBe('We are the Champions');
+      expect(res.body[1]).toHaveProperty('artistName');
+      expect(res.body[1].artistName).toBe('Queen');
+      expect(res.body[1]).toHaveProperty('genre');
+      expect(res.body[1].genre).toBe('Rock');
+      expect(res.body[1]).toHaveProperty('rating');
+      expect(res.body[1].rating).toBe(100);
```
Comment (Jonpatt92) [inline tests/requests/api/v1/favorites/index.spec.js]: Great improvement of our testing

**Pull request merged**